src/load/azext_load/data_plane/load_test_run/custom.py

```diff
@@ -0,0 +1,223 @@
+from azext_load.data_plane.utils.utils import (
+    create_or_update_test_run_body,
+    get_admin_data_plane_client,
+    get_test_run_id,
+    get_testrun_data_plane_client,
+)
+from azure.cli.core.azclierror import ValidationError
+from knack.log import get_logger
+import os, requests
+logger = get_logger(__name__)
+
+
+def create_test_run(
+    cmd,
+    load_test_resource,
+    test_id,
+    display_name=None,
+    existing_test_id=None,
+    description=None,
+    env=None,
+    secrets=None,
+    certificate=None,
+    resource_group_name=None,
+    no_wait=False,
+):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test_run_body = create_or_update_test_run_body(
+        test_id,
+        display_name=display_name,
+        description=description,
+        env=env,
+        secrets=secrets,
+        certificate=certificate,
+    )
+    logger.info("Creating test run with following request %s", test_run_body)
+    poller = client.begin_test_run(
+        test_run_id=get_test_run_id(),
+        body=test_run_body,
+        old_test_run_id=existing_test_id,
+    )
+    response = poller.polling_method()._initial_response
+    if not no_wait:
+        response = poller.result()
+    logger.info("Test run created with following response %s", response)
+    return response
+
+def get_test_run(cmd, load_test_resource, test_run_id, resource_group_name=None):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    logger.info("Getting test run %s", test_run_id)
+    return client.get_test_run(test_run_id=test_run_id)
+
+def update_test_run(
+    cmd,
+    test_run_id,
+    description,
+    load_test_resource,
+    resource_group_name=None,
+):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    # validate if test run exists
+    test_run_body = client.get_test_run(test_run_id=test_run_id)
```

Comment:
Do we throw an error for 404?

Reply:
Yes, SDK throws exception for everything except 2xx